### PR TITLE
Support fetching latest SNAPSHOT artifacts

### PR DESCRIPTION
**Pull request by alexjfisher**
RFC/WIP: DO NOT MERGE

This works for me and fixes
https://github.com/voxpupuli/puppet-archive/issues/282

Feedback and some volunteers to test welcome.

### CODE REVIEW

File: manifests/artifactory.pp [hunk 1]
```diff
@@ -72,24 +72,43 @@
   }
 
   if $url {
-    $file_url = $url
-    $sha1_url = regsubst($url, '/artifactory/', '/artifactory/api/storage/')
+    $maven2_data = archive::parse_artifactory_url($url)
```
Comment (alexjfisher) [inline manifests/artifactory.pp]: if `parse_artifactory_url` fails to parse the url, it returns `nil`

File: manifests/artifactory.pp [hunk 2]
```diff
@@ -72,24 +72,43 @@
...
+    if $maven2_data and $maven2_data['folder_iteg_rev'] == 'SNAPSHOT'{
```
Comment (alexjfisher) [inline manifests/artifactory.pp]: so if `maven2_data` is nil, the new code won't be run.

**Comment by bastelfreak** [comment]
looks good to me. I would like to merge & release this after a rebase

**Comment by alexjfisher** [comment]
I'd be fine letting it sit in master for a couple of weeks.  There's nothing else waiting for a release I don't think.

**Pull request merged**